vms/rpcchainvm/vm_client.go

```diff
@@ -352,8 +353,14 @@ func (vm *VMClient) SetState(ctx context.Context, state snow.State) error {
 }
 
 func (vm *VMClient) Shutdown(ctx context.Context) error {
+	// ensure the vm is alive before sending blocking shutdown RPC.
+	_, err := vm.healthCli.Check(ctx, &healthpb.HealthCheckRequest{Service: ""})
+	if err != nil {
+		return fmt.Errorf("shutdown failed vm has already been terminated: %w", err)
+	}
+
```

Comment:
TODO:  do health checks every 1s or 500ms and bail if for whatever reason this doesn't return